### ISSUE TITLE
Memory freeing

### ISSUE THREAD

**Issue by pawelkulig1** [LemurPwned/spintronics-visual#87]
I think you haven't checked if function removeDataObject is working. Python has garbage collector that deletes object when there is 0 refs, but another thing is that Python releases this memory when it wants not when we want that. Using del operator for example is considered bad practice. If we truly want to delete some object from mem, we should restart program or, delete all references to object and wait until it's deleted (It may never happen). I don't think there is proper solution for that because Python seems to be smarter. Fortunately it seems that garbage collector is doing good job at runtime end.

**Comment by pawelkulig1** [comment]
You are right but I doesn't change anything even if you set all this variables to None, memory is not released. Check it with memory profiler.

**Comment by LemurPwned** [comment]
Right, it doesnt now if the widgets are still running, we are just removing reference from doh which would be still there after widget is destroyed.

**Comment by pawelkulig1** [comment]
Closer but still not there :P

**Comment by pawelkulig1** [comment]
Of course problem is connected with deleting only reference

**Comment by LemurPwned** [comment]
What do you mean? I think I dont get it. What I meant is that we merely nullify reference in doh objects not outside. If reference exists outside then it should, but since doh is sort of global it must be cleaned manually, but that doest equal data deletion
